extensions.go

```diff
@@ -2,13 +2,83 @@ package csproto
 
 import (
 	"fmt"
+	"reflect"
 
 	gogo "github.com/gogo/protobuf/proto"
 	google "github.com/golang/protobuf/proto" //nolint: staticcheck // we're using this deprecated package intentionally
 	googlev2 "google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/reflect/protoreflect"
 )
 
+// Extension is a combined struct describing features from gogo, google v1 & v2, containing all
+// common descriptor values and the underlying extension value.
+// NOTE: Certain tags have been excluded as they are deprecated or non-existent or certain libraries
+type Extension struct {
```

Comment:
We should minimize the footprint of this struct to only the fields that are specifically required for some defined use case.  I'm thinking that only `Name` and `Field` are **required** and `Value` provides the underlying descriptor for more advanced usage (which will likely also be coupled to the underlying runtime).

Reply:
I'm quite happy to make that change ! 👏

Reply:
So, I've completely removed the `Extension` struct and just passed the values instead in trying to remove as much logic as possible. Hope this aligns with your feedback 🙇

Reply:
That's even better. :+1: